x/ecocredit/server/core/mint_batch_credits_test.go

```diff
@@ -0,0 +1,179 @@
+	balBefore, err := s.stateStore.BatchBalanceTable().Get(ctx, newAcc, batch.Key)
+	assert.ErrorContains(t, err, ormerrors.NotFound.Error())
+	balBefore = &api.BatchBalance{BatchKey: batch.Key, Address: newAcc}
+	supplyBefore, err := s.stateStore.BatchSupplyTable().Get(ctx, batch.Key)
+	assert.NilError(t, err)
+
+	mintTradable, mintRetired := math.NewDecFromInt64(10), math.NewDecFromInt64(10)
+	msg := core.MsgMintBatchCredits{
+		Issuer:     s.addr.String(),
+		BatchDenom: batch.Denom,
+		Issuance: []*core.BatchIssuance{
+			{Recipient: newAcc.String(), TradableAmount: mintTradable.String(), RetiredAmount: mintRetired.String(), RetirementJurisdiction: "US-OR"},
+		},
+		OriginTx: &core.OriginTx{
+			Typ: "Ethereum",
+			Id:  "210985091248",
+		},
+		Note: "bridged credits",
+	}
+
+	_, err = s.k.MintBatchCredits(ctx, &msg)
+	assert.NilError(t, err)
+
+	balAfter, err := s.stateStore.BatchBalanceTable().Get(ctx, newAcc, batch.Key)
+	assert.NilError(t, err)
+	supplyAfter, err := s.stateStore.BatchSupplyTable().Get(ctx, batch.Key)
+	assert.NilError(t, err)
+
+	assertCreditsMinted(t, balBefore, balAfter, supplyBefore, supplyAfter, mintTradable, mintRetired, math.NewDecFromInt64(0), 6)
+}
+
+func TestMintBatchCredits_Unauthorized(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	batch := setupMintBatchTest(s, true)
+	addr := sdk.AccAddress("foobar")
+
+	_, err := s.k.MintBatchCredits(s.ctx, &core.MsgMintBatchCredits{
+		Issuer:     addr.String(),
+		BatchDenom: batch.Denom,
+		Issuance:   nil,
+		OriginTx:   nil,
+		Note:       "",
+	})
+	assert.ErrorContains(t, err, "unauthorized")
+}
+
+func TestMintBatchCredits_ClosedBatch(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	batch := setupMintBatchTest(s, false)
+	addr := sdk.AccAddress("foobar")
+
+	_, err := s.k.MintBatchCredits(s.ctx, &core.MsgMintBatchCredits{
+		Issuer:     addr.String(),
+		BatchDenom: batch.Denom,
+		Issuance:   nil,
+		OriginTx:   nil,
+		Note:       "",
+	})
+	assert.ErrorContains(t, err, "credits cannot be minted in a closed batch")
+}
+
+func TestMintBatchCredits_NotFound(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	setupMintBatchTest(s, true)
+	addr := sdk.AccAddress("foobar")
+
+	_, err := s.k.MintBatchCredits(s.ctx, &core.MsgMintBatchCredits{
+		Issuer:     addr.String(),
+		BatchDenom: "C05-00000000-00000000-001",
+		Issuance:   nil,
+		OriginTx:   nil,
+		Note:       "",
+	})
+	assert.ErrorContains(t, err, ormerrors.NotFound.Error())
+}
+
+func setupMintBatchTest(s *baseSuite, open bool) *api.Batch {
+	ctx := s.ctx
+	_, _, batchDenom := s.setupClassProjectBatch(s.t)
+	batch, err := s.stateStore.BatchTable().GetByDenom(ctx, batchDenom)
+	assert.NilError(s.t, err)
+	batch.Open = open
+	assert.NilError(s.t, s.stateStore.BatchTable().Update(ctx, batch))
+	return batch
+}
+
+func assertCreditsMinted(t *testing.T, balBefore, balAfter *api.BatchBalance, supBefore, supAfter *api.BatchSupply, tradable, retired, escrowed math.Dec, precision uint32) {
```

Comment:
As mentioned above, a struct might be nice given there are nine parameters here.

Reply:
reduced it a bit here - https://github.com/regen-network/regen-ledger/pull/1059/commits/5268b06df0a06a0ce8a203e65b8fa2babec65c5b

passing the issuance as a whole rather than individual decimals. lmk if you think this should be further reduced

Reply:
That looks good to me.